arm-asm-s2s/arm-asm-s2s.json

```diff
@@ -0,0 +1,171 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "location": {
+            "type": "String",
+            "defaultValue": "Central US",
+            "allowedValues": [
+                "North Central US",
+                "East Asia",
+                "West US",
+                "Central US"
+            ]
+        },
+        "enableBgp": {
+            "type": "String",
+            "defaultValue": "false",
+            "allowedValues": [
+                "false"
+            ]
+        },
+        "gatewayType": {
+            "type": "String",
+            "defaultValue": "Vpn",
+            "allowedValues": [
+                "Vpn",
+                "ER"
+            ]
+        },
+        "vpnType": {
+            "type": "String",
+            "defaultValue": "RouteBased",
+            "allowedValues": [
+                "RouteBased",
+                "PolicyBased"
+            ]
+        },
+        "subscriptionId": {
+            "type": "string"
+        },
+        "localGatewayName": {
+            "type": "string"
+        },
+        "localGatewayIpAddress": {
+            "type": "string"
+        },
+        "localGatewayAddressPrefix": {
+            "type": "string"
+        },
+        "virtualNetworkName": {
```

Comment:
Parameters like VNET name, address prefix, subnet name etc. can become variables so that the number of parameters a user has to enter can be reduced. See other templates to guage what kind of parameters can become variables.